diff --git a/CI/full-build-macos.sh b/CI/full-build-macos.sh
index 80a73fc5b..85c718542 100755
--- a/CI/full-build-macos.sh
+++ b/CI/full-build-macos.sh
@@ -52,9 +52,6 @@ NPROC="${NPROC:-$(sysctl -n hw.ncpu)}"
 BUILD_DEPS=(
     "obs-deps ${MACOS_DEPS_VERSION:-${CI_DEPS_VERSION}}"
     "qt-deps ${QT_VERSION:-${CI_QT_VERSION}} ${MACOS_DEPS_VERSION:-${CI_DEPS_VERSION}}"
-    "cef ${CEF_BUILD_VERSION:-${CI_CEF_VERSION}}"
-    "vlc ${VLC_VERSION:-${CI_VLC_VERSION}}"
-    "sparkle ${SPARKLE_VERSION:-${CI_SPARKLE_VERSION}}"
 )
 
 if [ -n "${TERM-}" ]; then
@@ -270,17 +267,15 @@ configure_obs_build() {
     ensure_dir "${CHECKOUT_DIR}/${BUILD_DIR}"
 
     hr "Run CMAKE for OBS..."
-    cmake -DENABLE_SPARKLE_UPDATER=ON \
+    cmake \
         -DCMAKE_OSX_DEPLOYMENT_TARGET=${MIN_MACOS_VERSION:-${CI_MIN_MACOS_VERSION}} \
         -DDISABLE_PYTHON=ON  \
+        -DENABLE_SCRIPTING=0 \
+        -DDISABLE_PLUGINS=true \
         -DQTDIR="/tmp/obsdeps" \
         -DSWIGDIR="/tmp/obsdeps" \
         -DDepsPath="/tmp/obsdeps" \
-        -DVLCPath="${DEPS_BUILD_DIR}/vlc-${VLC_VERSION:-${CI_VLC_VERSION}}" \
-        -DBUILD_BROWSER=ON \
-        -DBROWSER_LEGACY="$(test "${CEF_BUILD_VERSION:-${CI_CEF_VERSION}}" -le 3770 && echo "ON" || echo "OFF")" \
         -DWITH_RTMPS=ON \
-        -DCEF_ROOT_DIR="${DEPS_BUILD_DIR}/cef_binary_${CEF_BUILD_VERSION:-${CI_CEF_VERSION}}_macosx64" \
         -DCMAKE_BUILD_TYPE="${BUILD_CONFIG}" \
         ..
 
