diff --git a/dlib/cmake_utils/find_blas.cmake b/dlib/cmake_utils/find_blas.cmake
index 806b9388..15fa3f46 100644
--- a/dlib/cmake_utils/find_blas.cmake
+++ b/dlib/cmake_utils/find_blas.cmake
@@ -198,7 +198,7 @@ if (UNIX OR MINGW)
       )
 
    if (NOT blas_found)
-      find_library(cblas_lib NAMES openblasp openblas PATHS ${extra_paths})
+      find_library(cblas_lib NAMES openblas PATHS ${extra_paths})
       if (cblas_lib)
          set(blas_libraries ${cblas_lib})
          set(blas_found 1)
